packages/amazon-cognito-identity-js/__tests__/StorageHelperRN.test.js

```diff
@@ -0,0 +1,126 @@
+describe('React native storage helper unit tests', () => {
+	describe('Accessor methods', () => {
+		beforeEach(() => {
+			jest.resetModules();
+		});
+
+		var items = {};
+		jest.mock('react-native', () => ({
+			AsyncStorage: {
+				setItem: jest.fn((item, value) => {
+					return new Promise((resolve, reject) => {
+						items[item] = value;
+						resolve(value);
+					});
+				}),
+				getItem: jest.fn((item, value) => {
+					return new Promise((resolve, reject) => {
+						resolve(items[item]);
+					});
+				}),
+				removeItem: jest.fn(item => {
+					return new Promise((resolve, reject) => {
+						resolve(delete items[item]);
+					});
+				}),
+			},
+		}));
+
+		const StorageHelper = require('../src/StorageHelper-rn.js').default;
+		const storageHelper = new StorageHelper();
+		test('setting an item', () => {
+			expect(
+				storageHelper.getStorage().setItem('testKey', 'testValue')
+			).toEqual('testValue');
+		});
+		test('getting an item', () => {
+			expect(storageHelper.getStorage().getItem('testKey')).toEqual(
+				'testValue'
+			);
+		});
+		test('removing an item', () => {
+			expect(storageHelper.getStorage().removeItem('testKey')).toEqual(true);
+		});
+		test('clearing the storage', () => {
+			storageHelper.getStorage().setItem('item1', 'value1');
+			expect(storageHelper.getStorage().clear()).toEqual({});
+		});
+	});
+
+	describe('Testing StorageHelper sync', () => {
+		beforeEach(() => {
+			jest.resetModules();
+		});
+		test('Happy case for syncing', () => {
+			var items = { key: 'value1', key2: 'value2' };
```

Comment:
Let's have `items` simulate the contents of AsyncStorage
```suggestion
			const items = {
				'@MemoryStorage:key': 'value1',
			};
```